x/mongo/driver/operation/drop_database.go

```diff
@@ -31,66 +30,31 @@ type DropDatabase struct {
 	deployment   driver.Deployment
 	selector     description.ServerSelector
 	writeConcern *writeconcern.WriteConcern
-	result       DropDatabaseResult
 	serverAPI    *driver.ServerAPIOptions
 }
 
-type DropDatabaseResult struct {
-	// The dropped database.
-	Dropped string
-}
-
-func buildDropDatabaseResult(response bsoncore.Document, srvr driver.Server) (DropDatabaseResult, error) {
-	elements, err := response.Elements()
-	if err != nil {
-		return DropDatabaseResult{}, err
-	}
-	ddr := DropDatabaseResult{}
-	for _, element := range elements {
-		switch element.Key() {
-		case "dropped":
-			var ok bool
-			ddr.Dropped, ok = element.Value().StringValueOK()
-			if !ok {
-				err = fmt.Errorf("response field 'dropped' is type string, but received BSON type %s", element.Value().Type)
-			}
-		}
-	}
-	return ddr, nil
-}
-
 // NewDropDatabase constructs and returns a new DropDatabase.
 func NewDropDatabase() *DropDatabase {
 	return &DropDatabase{}
 }
 
-// Result returns the result of executing this operation.
-func (dd *DropDatabase) Result() DropDatabaseResult { return dd.result }
-
-func (dd *DropDatabase) processResponse(info driver.ResponseInfo) error {
-	var err error
-	dd.result, err = buildDropDatabaseResult(info.ServerResponse, info.Server)
-	return err
-}
-
 // Execute runs this operations and returns an error if the operaiton did not execute successfully.
 func (dd *DropDatabase) Execute(ctx context.Context) error {
 	if dd.deployment == nil {
 		return errors.New("the DropDatabase operation must have a Deployment set before Execute can be called")
 	}
 
 	return driver.Operation{
-		CommandFn:         dd.command,
-		ProcessResponseFn: dd.processResponse,
```

Comment:
What happens if there's an error from `DropDatabase()`? Will the error not be returned now that there's no passed-in `ProcessResponseFn` when creating/executing the `Operation`?

Reply:
No, the only errors that ProcessResponseFn pass back are errors for the actual processing. Operation.decodeResult will separate out server errors before that, and those will get returned

Reply:
Ah that makes sense. Sounds good!